### ISSUE TITLE
iOS: Paypal here side loader doesn't open paypal here app.

### ISSUE THREAD

**Issue by EddieLukeAtmey** [paypal/here-sideloader-api-samples#35]
This happened just recently. I used the exactly same code as the sample provide, and used to have no problem with it. 

I'm not sure when, but about less than 1 month ago, a customer told us it no longer work. So, I test on my device (which was also working as expected before), and instead of popup to the paypal here app, it show the app store link to paypal here. Not sure what to do here. Please help.

This is the code I use to sideload anyway:

    NSString *encodedInvoice = [jsonInvoice stringByAddingPercentEscapesUsingEncoding:NSASCIIStringEncoding];

    NSString *encodedPaymentTypes = [@"cash,card,paypal" stringByAddingPercentEscapesUsingEncoding:NSASCIIStringEncoding];

    NSString *encodedReturnUrl = [@"myapp://handler?{result}Type={Type}&InvoiceId={InvoiceId}&Tip={Tip}&Email={Email}&TaxId={TxId}" stringByAddingPercentEscapesUsingEncoding:NSASCIIStringEncoding];

    NSString *pphUrlString = [NSString stringWithFormat:@"paypalhere://takePayment?accepted=%@&returnUrl=%@&invoice=%@&step=choosePayment",
                              encodedPaymentTypes, encodedReturnUrl, encodedInvoice];

**Comment by jaredegan** [comment]
Hi, that's a very strange behavior. Can you show where you actually open `pphUrlString`? Sorry but I have to ask, are you sure PayPal Here is installed? Have you launched the app on your device at least once?

Assuming your code and integration is correct, one workaround I would suggest is:
- Delete PayPal Here from your device
- Download PayPal Here from the app store
- Launch PayPal Here
- Try your integration again.

**Comment by EddieLukeAtmey** [comment]
Hi jaredegan, here is the code right after those I post above:

NSURL *pphUrl = [NSURL URLWithString:pphUrlString];

    NSLog(@"%@", pphUrlString);

    UIApplication *application = [UIApplication sharedApplication];
    if ([application canOpenURL:pphUrl]){
        [application openURL:pphUrl];
    }
    else {
        [application openURL:[NSURL URLWithString:@"itms://itunes.apple.com/us/app/paypal-here-for-ipad-accept/id607485062?mt=8"]];
        //                              itunes.apple.com/us/app/paypal-here/id505911015?mt=8"]]; // iphone link
    }

I haven't update this since it was working fine. But just suddenly it does not. Strange, right?

Let me try your suggestion. But my customer claimed that he has tried your work around and still not working.

**Comment by EddieLukeAtmey** [comment]
Now it's working again? I'm not sure what to do with this :/

![4cuol](https://cloud.githubusercontent.com/assets/10751944/10357627/6dbc9ccc-6dae-11e5-9624-87cc551de76c.jpg)

**Comment by rdantonio** [comment]
iOS 9 now requires that apps declare the query schemes that they want to open. Did you add this key to your info plist?

    <key>LSApplicationQueriesSchemes</key>
    <array>
      <string>paypalhere</string>
    </array>

I'm not sure how this will affect PPH though since they can technically open any URL scheme passed by the callback URL...

**Comment by jaredegan** [comment]
@rdantonio, just to clarify, that's a question for @EddieLukeAtmey right?

PayPal Here has not been delivered to the app store being built with the iOS 9 SDK yet, so any iOS 9 changes on our side would not affect the app as it is today.

I just did some experiments with this, deleting and reinstalling PayPal Here. Found some failures of Safari to open the URL, but I couldn't get it to be totally consistent. It seems that the initial opening of PPH is an unnecessary step. But I did notice one oddity:

- Deleted PPH.
- Open Safari, attempt to open a `paypalhere://` URL. Nothing happened as expected.
- Installed PPH from app store.
- Go back to Safari, attempt to open a `paypalhere://` URL. Nothing happened, unexpected.
- Force quitting Safari and reopening it then caused the `paypalhere://` URL to start working. It seems that the PPH app may need to be installed before the app that wants to call it is launched.

@EddieLukeAtmey, you reopened this issue after closing it, is there still a problem?

**Comment by rdantonio** [comment]
Correct, my comment was an offer of a solution for him (since he reopened it) as I had to put the same change in for it to work with our application.

The issue I found with our application was related to the canOpenUrl call, which will ALWAYS fail if that key is not included in the plist, therefore it would always redirect to the app store, even if the application is installed.

**Comment by EddieLukeAtmey** [comment]
@rdantonio I've just checked, yeah probably that was the cause. I haven't include that key to the info.plist. I'll fix this right away.

@jaredegan Thank you for your information. Also, yeah, I reopened the issue because I'm unsure if this will really work, then it turn out that I was testing on iOS 8.4. The real problem is probably as @rdantonio  said. 

So, I'll check this asap. As for now, I'll close the issue. If anything happen, I'll open it again.
Thank you guys.
